### PR TITLE
SALTO-5022: Adding service URLs for SF auto response rules

### PR DESCRIPTION
**Pull request by yelly**
Also doing some cleanup in `lightning_url_resolvers.ts`.

---

_Additional context for reviewer_
In SF there are exactly 2 instances of `AutoResponseRules`: Lead and Case. Adding a `_service_url` for them will allow us to support **Go To Service** for them.

---
_Release Notes_: 

_Salesforce Adapter_:
- Support for Go To Service for Auto Response Rules elements.

---
_User Notifications_: 
None.

### CODE REVIEW

File: packages/salesforce-adapter/src/elements_url_retreiver/lightning_url_resolvers.ts [hunk 1]
```diff
@@ -89,12 +92,14 @@ const METADATA_TYPE_TO_URI: Record<string, string> = {
 const getTypeIdentifier = async (
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/elements_url_retreiver/lightning_url_resolvers.ts:92]: After using `apiNameSync` this method can be converted to a sync one

**Review by tamtamirr — COMMENTED** [review]
Good Work! Added some comments 💬

File: packages/salesforce-adapter/src/elements_url_retreiver/lightning_url_resolvers.ts [hunk 2]
```diff
@@ -89,12 +92,14 @@ const METADATA_TYPE_TO_URI: Record<string, string> = {
...
   element?: Element,
 ): Promise<string | undefined> =>
-  element === undefined ? undefined : getInternalId(element) ?? apiName(element)
+  element === undefined
+    ? undefined
+    : getInternalId(element) ?? safeApiName(element)
 
 const getFieldIdentifier = async (element: Field): Promise<string> =>
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/src/elements_url_retreiver/lightning_url_resolvers.ts:99]: `safeApiName` is still async.

[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/src/elements_url_retreiver/lightning_url_resolvers.ts:99]: Done.

File: packages/salesforce-adapter/test/elements_url_retreiver.test.ts [hunk 3]
```diff
@@ -145,6 +145,24 @@ describe('lightningElementsUrlRetriever', () => {
...
+          }),
+          { fullName: 'Case' },
+        )
+        await expect(
+          elementUrlRetriever?.retrieveUrl(element),
+        ).resolves.toEqual(
+          new URL(
+            'https://salto5-dev-ed.lightning.force.com/lightning/setup/CaseResponses/home',
+          ),
+        )
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/test/elements_url_retreiver.test.ts:163]: Done. Also changed all the other tests.

File: packages/salesforce-adapter/src/elements_url_retreiver/lightning_url_resolvers.ts (hunk 1 above)
Comment (yelly) [reply packages/salesforce-adapter/src/elements_url_retreiver/lightning_url_resolvers.ts:92]: Done.

**Review by tamtamirr — APPROVED** [review]
LGTM!

**Pull request merged**